packages/snap/src/helpers/credential-helpers.ts

```diff
@@ -1,47 +1,362 @@
+};
+
+const getClaimIdFromVC = (credential: W3CCredential) => {
+  try {
+    const claimIdUrl = new URL(credential.id);
+
+    const pathNameParts = claimIdUrl.pathname.split('/');
+
+    return pathNameParts[pathNameParts.length - 1];
+  } catch (error) {
+    return credential.id;
+  }
+};
+
+const getClaimIdsFromOffer = (offer: ClaimOffer) => {
+  return offer.body.credentials.map((cred) => cred.id);
+};
+
+const loadAllCredentialsListPages = async <
+  V extends Record<string, unknown>,
+  Res extends
+    | GetAllVerifiableCredentialsQuery
+    | GetVerifiableCredentialsByClaimIdQuery
+    | GetVerifiableCredentialsByQueryHashQuery
+>(
+  request: DocumentNode,
+  variables: V,
+  ceramicProvider: CeramicProvider,
+): Promise<Res[]> => {
+  if (!ceramicProvider.client().did?.id) {
+    await ceramicProvider.auth();
+  }
+
+  const client = ceramicProvider.client();
+
+  const encryptedVerifiableCredentials: Res[] = [];
+
+  let endCursor: string | undefined | null = '';
+  let hasNextPage: boolean | undefined | null = false;
+
+  do {
+    const _variables: Record<string, unknown> = {
+      ...variables,
+      ...(endCursor && { after: endCursor }),
+    };
+
+    const { data, errors } = await client.execute<Res>(request, _variables);
+
+    if (errors) {
+      throw new TypeError(JSON.stringify(errors));
+    }
+
+    if (!data) {
+      throw new TypeError('No data returned from Ceramic');
+    }
+
+    encryptedVerifiableCredentials.push(data);
+
+    [endCursor, hasNextPage] = [
+      data?.verifiableCredentialIndex?.pageInfo?.endCursor,
+      data?.verifiableCredentialIndex?.pageInfo?.hasNextPage,
+    ];
+  } while (hasNextPage);
+
+  return encryptedVerifiableCredentials;
+};
+
+export const getAuthenticatedCeramicProvider = async (
+  pkHex?: string,
+  serverURL?: string,
+) => {
+  let privateKeyHex = pkHex ?? '';
+
+  if (!pkHex) {
+    const identityStorage = await getItemFromStore(StorageKeys.identity);
+
+    if (!identityStorage) {
+      throw new Error('Identity not created yet');
+    }
+
+    privateKeyHex = identityStorage.privateKeyHex;
+  }
+
+  const ceramicProvider = new CeramicProvider(privateKeyHex, serverURL);
+
+  await ceramicProvider.auth();
+
+  return ceramicProvider;
 };
 
+export class VCManager {
+  ceramicProvider: CeramicProvider;
+
+  private mmKeyHex: string;
+
+  constructor(ceramicProvider: CeramicProvider, mmKeyHex: string) {
+    this.ceramicProvider = ceramicProvider;
+    this.mmKeyHex = mmKeyHex;
+  }
+
+  static async create(pkHex?: string, serverURL?: string) {
+    const entropy = await snap.request({
+      method: 'snap_getEntropy',
+      params: { version: 1 },
+    });
+    const keyHex = entropy.startsWith('0x') ? entropy.substring(2) : entropy;
+
+    return new VCManager(
+      await getAuthenticatedCeramicProvider(pkHex, serverURL),
+      keyHex,
+    );
+  }
+
+  private personalHashStr(str: string) {
+    return sha256(Buffer.from(str + _SALT + this.mmKeyHex));
```

Comment:
You are supposed to pass the salt to `getEntropy()` call, not to concat it to users private key